go-controller/pkg/ovndbmanager/ovndbmanager.go

```diff
@@ -0,0 +1,214 @@
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		ensureOvnDBState(util.OvnNbdbLocation, stopCh)
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		ensureOvnDBState(util.OvnSbdbLocation, stopCh)
+	}()
+	<-stopCh
+	klog.Info("Shutting down db checker")
+	wg.Wait()
+	klog.Info("Shut down db checker")
+}
+
+func ensureOvnDBState(db string, stopCh <-chan struct{}) {
+	ticker := time.NewTicker(60 * time.Second)
+	klog.Infof("Starting ensure routine for Raft db: %s", db)
+	_, _, err := util.RunOVSDBTool("db-is-standalone", db)
+	if err == nil {
+		klog.Info("The db is running in standalone mode")
+		return
+	}
+
+	for {
+		select {
+		case <-ticker.C:
+			klog.V(5).Infof("Ensure routines for Raft db: %s kicked off by ticker", db)
+			ensureLocalRaftServerID(db)
+			ensureClusterRaftMembership(db)
+			ensureDBHealth(db)
+		case <-stopCh:
+			ticker.Stop()
+			return
+		}
+	}
+}
+
+// ensureLocalRaftServerID is used to ensure there is no stale member in the Raft cluster with our address
+func ensureLocalRaftServerID(db string) {
+	var dbName string
+	var appCtl func(args ...string) (string, string, error)
+
+	if strings.Contains(db, "ovnnb") {
+		dbName = "OVN_Northbound"
+		appCtl = util.RunOVNNBAppCtl
+	} else {
+		dbName = "OVN_Southbound"
+		appCtl = util.RunOVNSBAppCtl
+	}
+
+	out, stderr, err := util.RunOVSDBTool("db-sid", db)
+	if err != nil {
+		klog.Warningf("Unable to get db server ID for: %s, stderr: %v, err: %v", db, stderr, err)
+		return
+	}
+	if len(out) < 4 {
+		klog.Errorf("Invalid db id found: %s for db: %s", out, db)
+		return
+	}
+	// server ID in raft membership is only first 4 char prefix
+	serverID := out[:4]
+	out, stderr, err = appCtl("cluster/status", dbName)
+	if err != nil {
+		klog.Warningf("Unable to get cluster status for: %s, stderr: %v, err: %v", db, stderr, err)
+		return
+	}
+	r, _ := regexp.Compile(`Address: *((ssl|tcp):[?[a-z0-9.:]+]?)`)
+	matches := r.FindStringSubmatch(out)
+	if len(matches) < 2 {
+		klog.Warningf("Unable to parse Address for db: %s, output: %s", db, out)
+		return
+	}
+	addr := matches[1]
+	// look for current servers in raft cluster with the same address
+	r, _ = regexp.Compile("([a-z0-9]{4}) at " + addr)
+	members := r.FindAllStringSubmatch(out, -1)
+	for _, member := range members {
+		if len(member) < 2 {
+			klog.Warningf("Unable to find server id submatch in %s", member)
+			return
+		}
+		if member[1] != serverID {
+			// stale entry found for this node with same adddress, need to kick
+			klog.Infof("Previous stale member found: %s...kicking", member[1])
+			_, stderr, err = appCtl("cluster/kick", dbName, member[1])
+			if err != nil {
+				klog.Errorf("Error while kicking old Raft member: %s, for address: %s in db: %s,"+
+					"stderr: %v, error: %v", member[1], addr, db, stderr, err)
+			}
+		}
+	}
+}
+
+// ensureClusterRaftMembership ensures there are no unknown members in the current Raft cluster
```

Comment:
why not only do this on leader?

Reply:
Yes, we could potentially optimize, but it doesn't hurt to run this on each db node for now so that if for whatever reason the current raft leader is down, the other two db-checker instances can still prune unknown members. Otherwise, we will have to add logic to keep monitoring if the local db instance is the current leader or not.

Reply:
Ok. But I believe the membership update also needs the raft concensus which requires majority agreement.  In your example, I don't think the other two db-check instances can prune unknown member without existence of the leader.

Reply:
Isn't one of the other two members going to become the raft leader in this case?

Reply:
Yes, that is exactly what I meant, that you can just do it on the leader and it would assue the same result. But as you said, it does not seem to hurt with your current code, I am ok with what you have.